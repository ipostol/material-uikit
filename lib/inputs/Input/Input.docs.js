'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputDocs = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkupComponents = require('react-markup-components');

var _SearchSmall = require('../../icons/SearchSmall');

var _SearchSmall2 = _interopRequireDefault(_SearchSmall);

var _Input = require('./Input');

var _Input2 = _interopRequireDefault(_Input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var InputDocs = exports.InputDocs = function InputDocs() {
  return _react2.default.createElement(
    _reactMarkupComponents.Skin,
    null,
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { defaultValue: '10pm' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { defaultValue: 'with info', info: 'Amount of \xA3234 incl. VAT will be charged from card' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { defaultValue: '10pm', label: 'Label', optional: true })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { icon: _react2.default.createElement(_SearchSmall2.default, { style: { display: 'block' } }), defaultValue: '10pm' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { label: 'Label', error: 'some error', defaultValue: '10pm' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { defaultValue: '10pm', secondary: true, label: 'Label' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { defaultValue: '10pm', icon: _react2.default.createElement(_SearchSmall2.default, null), secondary: true, label: 'Label' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { defaultValue: '10pm', error: 'Some error', secondary: true, label: 'Label' })
    ),
    _react2.default.createElement('br', null),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(_Input2.default, { disabled: true, secondary: true, defaultValue: '10pm' })
    )
  );
};

exports.default = InputDocs;