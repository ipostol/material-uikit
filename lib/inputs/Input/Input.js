'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Label = require('../../informers/Label');

var _Label2 = _interopRequireDefault(_Label);

var _Input = require('./Input.scss');

var _Input2 = _interopRequireDefault(_Input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Input = function (_PureComponent) {
  _inherits(Input, _PureComponent);

  function Input() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Input);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Input.__proto__ || Object.getPrototypeOf(Input)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      isFocused: false,
      clearedByFocus: false
    }, _this.onFocus = function (e) {
      var _this$props = _this.props,
          onFocus = _this$props.onFocus,
          onChange = _this$props.onChange,
          clearByFocus = _this$props.clearByFocus;


      _this.setState({ isFocused: true });

      if (clearByFocus && !_this.state.clearedByFocus) {

        _this.setState({ clearedByFocus: true });
        onChange(_extends({}, e, { target: _extends({}, e.target, { value: '' }) }));
      }

      onFocus(e);
    }, _this.onBlur = function (e) {
      var onBlur = _this.props.onBlur;


      _this.setState({ isFocused: false });

      onBlur(e);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Input, [{
    key: 'render',
    value: function render() {
      var _this2 = this,
          _cn;

      var _props = this.props,
          inputClassName = _props.inputClassName,
          className = _props.className,
          label = _props.label,
          error = _props.error,
          size = _props.size,
          icon = _props.icon,
          secondary = _props.secondary,
          disabled = _props.disabled,
          info = _props.info,
          optional = _props.optional,
          clearByFocus = _props.clearByFocus,
          rest = _objectWithoutProperties(_props, ['inputClassName', 'className', 'label', 'error', 'size', 'icon', 'secondary', 'disabled', 'info', 'optional', 'clearByFocus']);

      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)(_Input2.default.inputWrap, className) },
        _react2.default.createElement(
          _Label2.default,
          {
            isFocused: this.state.isFocused,
            optional: optional,
            isError: error
          },
          label
        ),
        icon ? _react2.default.createElement(
          'div',
          { className: _Input2.default.icon },
          icon
        ) : null,
        _react2.default.createElement('input', _extends({
          ref: function ref(_ref2) {
            return _this2.input = _ref2;
          }
        }, rest, {
          onBlur: this.onBlur,
          onFocus: this.onFocus,
          disabled: disabled,
          className: (0, _classnames2.default)(_Input2.default.general, (_cn = {}, _defineProperty(_cn, _Input2.default.error, error), _defineProperty(_cn, _Input2.default.disabled, disabled), _defineProperty(_cn, _Input2.default.iconPadding, !!icon), _defineProperty(_cn, _Input2.default.secondary, secondary), _cn), inputClassName, _Input2.default[size])
        })),
        error && _react2.default.createElement(
          'div',
          { className: _Input2.default.warning },
          error
        ),
        info && _react2.default.createElement(
          'div',
          { className: _Input2.default.info },
          info
        )
      );
    }
  }]);

  return Input;
}(_react.PureComponent);

Input.propTypes = {
  size: _propTypes2.default.string,
  label: _propTypes2.default.string,
  optional: _propTypes2.default.bool,
  className: _propTypes2.default.string,
  inputClassName: _propTypes2.default.string,
  error: _propTypes2.default.node,
  secondary: _propTypes2.default.bool,
  icon: _propTypes2.default.node,
  disabled: _propTypes2.default.bool,
  onFocus: _propTypes2.default.func,
  onBlur: _propTypes2.default.func,
  onChange: _propTypes2.default.func,
  info: _propTypes2.default.string,
  clearByFocus: _propTypes2.default.bool
};

Input.defaultProps = {
  type: 'text',
  size: 'l',
  onFocus: function onFocus() {},
  onBlur: function onBlur() {},
  onChange: function onChange() {}
};

exports.default = Input;