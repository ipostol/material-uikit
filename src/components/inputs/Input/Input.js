import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Label from '../../informers/Label';
import styles from './Input.scss';

class Input extends PureComponent {

  state = {
    isFocused: false,
    clearedByFocus: false,
  };

  onFocus = (e) => {

    const { onFocus, onChange, clearByFocus } = this.props;

    this.setState({ isFocused: true });

    if (clearByFocus && !this.state.clearedByFocus) {

      this.setState({ clearedByFocus: true });
      onChange({ ...e, target: { ...e.target, value: '' } });

    }

    onFocus(e);

  };

  onBlur = (e) => {

    const { onBlur } = this.props;

    this.setState({ isFocused: false });

    onBlur(e);

  };

  render() {

    const {
      inputClassName,
      className,
      label,
      error,
      size,
      icon,
      secondary,
      disabled,
      info,
      optional,
      clearByFocus,
      ...rest
    } = this.props;

    return (
      <div className={cn(styles.inputWrap, className)}>
        <Label
          isFocused={this.state.isFocused}
          optional={optional}
          isError={error}
        >
          {label}
        </Label>
        {icon ? <div className={styles.icon}>{icon}</div> : null}
        <input
          ref={ref => (this.input = ref)}
          {...rest}
          onBlur={this.onBlur}
          onFocus={this.onFocus}
          disabled={disabled}
          className={
            cn(styles.general, {
              [styles.error]: error,
              [styles.disabled]: disabled,
              [styles.iconPadding]: !!icon,
              [styles.secondary]: secondary,
            }, inputClassName, styles[size])
          }
        />
        {error && <div className={styles.warning}>{error}</div>}
        {info && <div className={styles.info}>{info}</div>}
      </div>
    );

  }
}

Input.propTypes = {
  size: PropTypes.string,
  label: PropTypes.string,
  optional: PropTypes.bool,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  error: PropTypes.node,
  secondary: PropTypes.bool,
  icon: PropTypes.node,
  disabled: PropTypes.bool,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  info: PropTypes.string,
  clearByFocus: PropTypes.bool,
};

Input.defaultProps = {
  type: 'text',
  size: 'l',
  onFocus() {},
  onBlur() {},
  onChange() {},
};

export default Input;
