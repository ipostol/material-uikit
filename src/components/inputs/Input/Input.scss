@import '~bidstack-theme/theme';

:-ms-input-placeholder { color: $fontColorLight3; }
::-moz-placeholder { color: $fontColorLight3; }
::-webkit-input-placeholder { color: $fontColorLight3; }
:-moz-placeholder { color: $fontColorLight3; }

.general {
  @extend .body1;

  font-family: 'Open sans', sans-serif;
  outline: none;
  display: inline-flex;
  justify-content: flex-start;
  width: 100%;
  box-sizing: border-box;
  caret-color: $primeColor;
  color: $fontDark;
  border: none;
  border-bottom: 2px solid #d3d6e1;

  &:hover {
    cursor: text;
  }

  &:focus {
    color: $fontDark;
    border-bottom: 2px solid $primeColor;
  }

  &.m {
    height: 36px;
  }

  &.l {
    height: 45px;
  }
}

.secondary {
  border: 1px solid $borderColor;
  box-sizing: border-box;
  border-radius: 4px;
  background-color: $white3;
  height: 36px !important;
  padding: 6px 9px;

  &:hover {
    cursor: text;
  }

  &:focus {
    border: 1px solid $primeColor;
    box-shadow: 0 0 0 1px $primeColor;
  }

  &.disabled {
    border: 1px dashed #bdbdbd;
  }

  &.error:focus {
    box-shadow: none;
    border: 1px solid $red;
  }
}

.disabled {
  cursor: not-allowed !important;
  color: $fontLight;
  user-select: none;
}

.error {
  border-color: $fontRed;
  caret-color: $fontRed;

  &:focus {
    border-bottom: 2px solid $fontRed;
  }
}

.info {
  @extend .small1;

  margin-top: 9px;
  color: $fontLight;
}

.warning {
  @extend .small1;

  color: $red;
  margin-top: 3px;
}

.inputWrap {
  position: relative;
  width: 100%;

  & + & {
    margin-top: 32px;
  }
}

.iconPadding {
  padding-left: 34px;
}

.icon {
  position: absolute;
  left: 0;
  top: 50%;
  transform: translateY(-50%);
  margin-left: 8px;
}
