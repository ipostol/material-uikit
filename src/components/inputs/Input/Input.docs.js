import React from 'react';
import { Skin } from 'react-markup-components';
import SearchIcon from '../../icons/SearchSmall';
import Input from './Input';

export const InputDocs = () =>
  <Skin>
    <div>
      <Input defaultValue="10pm" />
    </div>
    <br />
    <div>
      <Input defaultValue="with info" info="Amount of Â£234 incl. VAT will be charged from card" />
    </div>
    <br />
    <div>
      <Input defaultValue="10pm" label="Label" optional />
    </div>
    <br />
    <div>
      <Input icon={<SearchIcon style={{ display: 'block' }} />} defaultValue="10pm" />
    </div>
    <br />
    <div>
      <Input label="Label" error="some error" defaultValue="10pm" />
    </div>
    <br />
    <div>
      <Input defaultValue="10pm" secondary label="Label" />
    </div>
    <br />
    <div>
      <Input defaultValue="10pm" icon={<SearchIcon />} secondary label="Label" />
    </div>
    <br />
    <div>
      <Input defaultValue="10pm" error="Some error" secondary label="Label" />
    </div>
    <br />
    <div>
      <Input disabled secondary defaultValue="10pm" />
    </div>
  </Skin>;

export default InputDocs;
