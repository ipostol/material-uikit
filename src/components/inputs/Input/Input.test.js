import React from 'react';
import { shallow, mount } from 'enzyme';
import Input from './Input';

describe('Input component', () => {

  it('should render correctly', () => {

    const props = {};

    const wrapper = shallow(
      <Input {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('should render correctly icon', () => {

    const props = {
      icon: <div />,
      error: 'error',
    };

    const wrapper = shallow(
      <Input {...props} />,
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('should mount correctly', () => {

    const props = {
      icon: <div />,
      error: 'error',
      onBlur() {},
    };

    const wrapper = mount(
      <Input {...props} />,
    );

    wrapper.find('input').simulate('blur');

    expect(wrapper).toMatchSnapshot();

  });

});
