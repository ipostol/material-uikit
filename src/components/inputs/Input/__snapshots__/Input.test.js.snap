// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should mount correctly 1`] = `
<Input
  error="error"
  icon={<div />}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  size="l"
  type="text"
>
  <div
    className="inputWrap"
  >
    <Label
      isError="error"
      isFocused={false}
    />
    <div
      className="icon"
    >
      <div />
    </div>
    <input
      className="general error iconPadding l"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
    <div
      className="warning"
    >
      error
    </div>
  </div>
</Input>
`;

exports[`Input component should render correctly 1`] = `
<div
  className="inputWrap"
>
  <Label
    isFocused={false}
  />
  <input
    className="general l"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Input component should render correctly icon 1`] = `
<div
  className="inputWrap"
>
  <Label
    isError="error"
    isFocused={false}
  />
  <div
    className="icon"
  >
    <div />
  </div>
  <input
    className="general error iconPadding l"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
  <div
    className="warning"
  >
    error
  </div>
</div>
`;
